bson/src/main/org/bson/json/JsonReader.java

```diff
@@ -1208,6 +1208,7 @@ private BsonBinary visitUuidExtendedJson() {
         String uuidString = readStringFromExtendedJson();
         verifyToken(JsonTokenType.END_OBJECT);
         try {
+            UuidStringValidator.validate(uuidString);
```

Comment:
Q: Once we've done all that validation shouldn't we just create the UUID, rather than push it through the validation in UUID.fromString again?

Reply:
I thought about it, but didn't want to take over that responsibility.  We could always do it later if there are complaints about performance.